### PR TITLE
Create DriverState

### PR DESCRIPTION
**Pull request by danieljanes**
- [x] Merge #1546 first

### CODE REVIEW

File: src/proto/flwr/proto/driver.proto [hunk 1]
```diff
@@ -24,20 +24,23 @@ service Driver {
...
 
-// GetResults messages
-message GetResultsRequest { repeated uint64 task_ids = 1; }
-message GetResultsResponse { repeated Result results = 1; }
+// PullTaskRes messages
+message PullTaskResRequest {
+  uint64 node_id = 1;
+  repeated string task_ids = 2;
+}
+message PullTaskResResponse { repeated TaskRes task_res_set = 1; }
```
Comment (tanertopal) [inline src/proto/flwr/proto/driver.proto:46]:
```suggestion
message PullTaskResResponse { repeated TaskRes task_res_list = 1; }
```

File: src/proto/flwr/proto/fleet.proto [hunk 2]
```diff
@@ -20,17 +20,32 @@ package flwr.proto;
...
-message GetTasksRequest {}
-message GetTasksResponse { repeated Task task = 1; }
+// PullTaskIns messages
+message PullTaskInsRequest {
+  uint64 node_id = 1;
+  repeated string task_ids = 2;
+}
+message PullTaskInsResponse {
+  Reconnect reconnect = 1;
+  repeated TaskIns task_ins_set = 2;
```
Comment (tanertopal) [inline src/proto/flwr/proto/fleet.proto:41]:
```suggestion
  repeated TaskIns task_ins_list = 2;
```

File: src/proto/flwr/proto/fleet.proto [hunk 3]
```diff
@@ -20,17 +20,32 @@ package flwr.proto;
...
+}
+
+// PushTaskRes messages
+message PushTaskResRequest { repeated TaskRes task_res_set = 1; }
```
Comment (tanertopal) [inline src/proto/flwr/proto/fleet.proto:45]:
```suggestion
message PushTaskResRequest { repeated TaskRes task_res_list = 1; }
```

File: src/py/flwr/server/driver/driver_servicer.py [hunk 4]
```diff
@@ -56,12 +61,83 @@ def PushTaskIns(
...
         """Pull a set of TaskRes."""
         log(INFO, "DriverServicer.PullTaskRes")
 
-        return PullTaskResResponse(task_res_set=[])
+        # Convert each task_id str to UUID
+        task_ids: Set[UUID] = {UUID(task_id) for task_id in request.task_ids}
+
+        # Read from state
+        task_res_set: List[TaskRes] = self.driver_state.get_task_res(
+            task_ids=task_ids, limit=10
```
Comment (tanertopal) [inline src/py/flwr/server/driver/driver_servicer.py:92]: Why 10?

File: src/py/flwr/server/driver/state_test.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+from datetime import datetime, timezone
+from uuid import uuid4
+
+from flwr.proto.node_pb2 import Node
+from flwr.proto.task_pb2 import Task, TaskIns
+from flwr.server.driver.state import DriverState
+
+
+def test_get_task_ins_empty() -> None:
+    """."""
```
Comment (tanertopal) [inline src/py/flwr/server/driver/state_test.py:27]: Why just "."?

File: src/py/flwr/server/driver/state_test.py [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+        node_id=1,
+        limit=10,
+    )
+
+    # Assert
+    assert not task_ins_set
+
+
+def test_get_task_res_empty() -> None:
+    """."""
```
Comment (tanertopal) [inline src/py/flwr/server/driver/state_test.py:43]: Why just "."?

File: src/py/flwr/server/driver/driver_servicer.py (hunk 4 above)
Comment (danieljanes) [reply src/py/flwr/server/driver/driver_servicer.py]: Good catch - I removed the limit, it didn't make sense here

**Review by tanertopal — APPROVED** [review]

**Pull request merged**